pkg/apis/api/types.go

```diff
@@ -817,23 +818,40 @@ type ComponentReportRequestAdvancedOptions struct {
 }
 
 type ComponentTestStatus struct {
-	TestName     string
-	TestSuite    string
-	Component    string
-	Capabilities []string
-	Variants     []string
-	TotalCount   int
-	SuccessCount int
-	FlakeCount   int
+	TestName     string   `json:"test_name"`
+	TestSuite    string   `json:"test_suite"`
+	Component    string   `json:"component"`
+	Capabilities []string `json:"capabilities"`
+	Variants     []string `json:"variants"`
+	TotalCount   int      `json:"total_count"`
+	SuccessCount int      `json:"success_count"`
+	FlakeCount   int      `json:"flake_count"`
+}
+
+type ComponentReportTestStatus struct {
+	BaseStatus   map[ComponentTestIdentification]ComponentTestStatus `json:"base_status"`
+	SampleStatus map[ComponentTestIdentification]ComponentTestStatus `json:"sample_status"`
+	GeneratedAt  *time.Time                                          `json:"generated_at"`
 }
 
 type ComponentTestIdentification struct {
-	TestID       string
-	Network      string
-	Upgrade      string
-	Arch         string
-	Platform     string
-	FlatVariants string
+	TestID       string `json:"test_id"`
+	Network      string `json:"network"`
+	Upgrade      string `json:"upgrade"`
+	Arch         string `json:"arch"`
+	Platform     string `json:"platform"`
+	FlatVariants string `json:"flat_variants"`
+}
+
+// implement encoding.TextMarshaler for json map key marshalling support
+func (s ComponentTestIdentification) MarshalText() (text []byte, err error) {
+	type t ComponentTestIdentification
+	return json.Marshal(t(s))
+}
+
+func (s *ComponentTestIdentification) UnmarshalText(text []byte) error {
+	type t ComponentTestIdentification
+	return json.Unmarshal(text, (*t)(s))
 }
```

Comment:
Are these needed? The struct should just be serializable on it's own.

Reply:
Yes I was getting errors [here](https://github.com/openshift/sippy/blob/9dfe15eea256c6a54d72bca9d0a56b0c532e319c/pkg/api/utils.go#L55) found this [issue](https://stackoverflow.com/questions/55335296/problem-with-marshal-unmarshal-when-key-of-map-is-a-struct) added a brief comment [here](https://github.com/openshift/sippy/blob/9dfe15eea256c6a54d72bca9d0a56b0c532e319c/pkg/apis/api/types.go#L846)